src/libraries/Microsoft.Extensions.Logging.Abstractions/gen/LoggerMessageGenerator.Parser.cs

```diff
@@ -632,6 +632,30 @@ private static string GenerateClassName(TypeDeclarationSyntax typeDeclaration)
 
                 bool onMostDerivedType = true;
 
+#if ROSLYN4_8_OR_GREATER
```

Comment:
>ROSLYN4_8_OR_GREATER [](http://example.com/codeflow?start=4&length=20)

I am seeing we are defining such constants in the csproj [https://github.com/dotnet/runtime/blob/017593d90781d797df8b2241f6d1f83c236c442b/src/libraries/Microsoft.Extensions.Logging.Abstractions/gen/Microsoft.Extensions.Logging.Generators.Roslyn4.4.csproj#L6](https://github.com/dotnet/runtime/blob/017593d90781d797df8b2241f6d1f83c236c442b/src/libraries/Microsoft.Extensions.Logging.Abstractions/gen/Microsoft.Extensions.Logging.Generators.Roslyn4.4.csproj#L6)

Could you try to use `ROSLYN4_4_OR_GREATER` instead and look if this will work? I expect we use recent Roslyn which I expect it will work.

Reply:
That didn't work with the previous version of this pull request, since the `ParameterList` property on `ClassDeclarationSyntax` isn't available until Roslyn 4.8, but it may be possible now that this uses the constructor parameters on the class symbol instead. Let me give it a try!

Reply:
Look at my suggestion https://github.com/dotnet/runtime/pull/101660#discussion_r1585301628. I am asking if we can do that instead of creating a new version of the library just to match the desired Roslyn version.